### PR TITLE
Automated ref finder for `asap-alchemy`

### PR DESCRIPTION
**Pull request by JenkeScheen**
## Description
For `asap-alchemy` we don't want to have to manually find a crystal structure to use as reference for docking every time we build a network. Instead, we want to find this automatically (https://github.com/choderalab/asapdiscovery/issues/918).

Caveats:
- we typically `prep` before we `dock`, but to find the correct reference we need the ligands, so this jumbles up the order a bit. I opted for loading the ligands during `prep`, then just re-loading them during `dock`. This isn't very elegant but has minimal overhead.
- references may/may not be compatible with `OpenMM` for a variety of reasons. I built in a check that makes sure we're moving forward with a compatible reference.
- I'm a horrible person so the code is a bit monolithic in its current state; @hmacdope should advise on how to move some things around, particularly in [this section](https://github.com/choderalab/asapdiscovery/pull/938/files#diff-e0af4b3e05dcd7e52116cd7e7fc0f91862414d4a9e628074511f00e732737315R389) and [this one](https://github.com/choderalab/asapdiscovery/pull/938/files#diff-e0af4b3e05dcd7e52116cd7e7fc0f91862414d4a9e628074511f00e732737315R371).
- I accidentally pushed to another branch (https://github.com/choderalab/asapdiscovery/pull/919) so I had to fix all that which resulted in this PR having a large bulk commit rather than incremental ones. 
- @jthorton is adding a `AlchemyPrepWorkflow` that the workflow in this PR could potentially migrated into once https://github.com/choderalab/asapdiscovery/pull/919 is merged

## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

### CODE REVIEW

**Comment by JenkeScheen** [comment]
@hmacdope can I grab a quick review? Feel free to just give some pointers on where things should be dropped if you think the current placing is off.

File: asapdiscovery-workflows/asapdiscovery/workflows/prep_workflows/protein_prep.py [hunk 1]
```diff
@@ -276,3 +322,139 @@ def protein_prep_workflow(inputs: ProteinPrepInputs):
...
+from openff.toolkit import ForceField, Topology
+from openmm import unit as openmm_unit
+
+
+### SOMETHING GOING WRONG HERE - NO BOX?!
+def create_protein_only_system(pdb_path):
+    # attempt to make an OpenMM system with the prepped protein.
+    # follows https://docs.openforcefield.org/en/latest/examples/openforcefield/openff-toolkit/toolkit_showcase/toolkit_showcase.html
+    top = Topology.from_pdb(pdb_path)
+    sage_ff14sb = ForceField("openff-2.1.0.offxml", "ff14sb_off_impropers_0.0.3.offxml")
```
Comment (JenkeScheen) [inline asapdiscovery-workflows/asapdiscovery/workflows/prep_workflows/protein_prep.py:430]: @jthorton notes we should use the FFs we use in production forr `asap-alchemy`

Comment (jthorton) [reply asapdiscovery-workflows/asapdiscovery/workflows/prep_workflows/protein_prep.py:430]: The default force fields can be found [here](https://github.com/OpenFreeEnergy/gufe/blob/29f98ec0227ed9f30ea6b84096dc13818a9ac8d3/gufe/settings/models.py#L161-L166). I think we just need to make an openMM system using the topology and the force fields listed and provided we don't hit an exception the pdb should be valid. Something like this should work 
```python
forcefield = openmm.ForceField([
        "amber/ff14SB.xml",  # ff14SB protein force field
        "amber/tip3p_standard.xml",  # TIP3P and recommended monovalent ion parameters
        "amber/tip3p_HFE_multivalent.xml",  # for divalent ions
        "amber/phosaa10.xml",  # Handles THE TPO
    ])
pdb = openmm.app.PDBFile(pdb_path)
try:
    _ = forcefield.createSystem(pdb.topology)
except Exception as e:
    raise InvalidStructureError from e
```

**Comment by hmacdope** [comment]
Closing in favour of #966

**Pull request closed without merging**